examples/tutorial/vertical_federated_learning/cifar10-splitnn/src/splitnn/cifar10_learner_splitnn.py

```diff
@@ -0,0 +1,614 @@
+            self.log_error(fl_ctx, f"expect model to be torch.nn.Module but got {type(self.model)}: {self.model}")
+            return
+        if self.model is None:
+            self.log_error(fl_ctx, f"Model wasn't built correctly! It is {self.model}")
+            return
+        self.log_info(fl_ctx, f"Running model {self.model}")
+
+    def initialize(self, parts: dict, fl_ctx: FLContext):
+        t_start = timer()
+        self._get_model(fl_ctx=fl_ctx)
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self.model = self.model.to(self.device)
+        self.optimizer = optim.SGD(self.model.parameters(), lr=self.lr, momentum=0.9)
+        self.criterion = torch.nn.CrossEntropyLoss()
+
+        self.transform_train = transforms.Compose(
+            [
+                transforms.ToTensor(),
+                transforms.ToPILImage(),
+                transforms.Pad(4, padding_mode="reflect"),
+                transforms.RandomCrop(32),
+                transforms.RandomHorizontalFlip(),
+                transforms.ToTensor(),
+                transforms.Normalize(
+                    mean=[x / 255.0 for x in [125.3, 123.0, 113.9]],
+                    std=[x / 255.0 for x in [63.0, 62.1, 66.7]],
+                ),
+            ]
+        )
+        self.transform_valid = transforms.Compose(
+            [
+                transforms.ToTensor(),
+                transforms.Normalize(
+                    mean=[x / 255.0 for x in [125.3, 123.0, 113.9]],
+                    std=[x / 255.0 for x in [63.0, 62.1, 66.7]],
+                ),
+            ]
+        )
+
+        self.app_root = fl_ctx.get_prop(FLContextKey.APP_ROOT)
+        self.client_name = fl_ctx.get_identity_name()
+        self.split_id = self.model.get_split_id()
+        self.log_info(fl_ctx, f"Running `split_id` {self.split_id} on site `{self.client_name}`")
+
+        if self.split_id == 0:  # data side
+            data_returns = "image"
+        elif self.split_id == 1:  # label side
+            data_returns = "label"
+        else:
+            raise ValueError(f"Expected split_id to be '0' or '1' but was {self.split_id}")
+
+        if self.intersection_file is not None:
+            _intersect_indices = np.loadtxt(self.intersection_file)
+        else:
+            _intersect_indices = None
+        self.train_dataset = CIFAR10SplitNN(
+            root=self.dataset_root,
+            train=True,
+            download=True,
+            transform=self.transform_train,
+            returns=data_returns,
+            intersect_idx=_intersect_indices,
+        )
+
+        self.valid_dataset = CIFAR10SplitNN(
+            root=self.dataset_root,
+            train=False,
+            download=False,
+            transform=self.transform_valid,
+            returns=data_returns,
+            intersect_idx=None,  # TODO: support validation intersect indices
+        )
```

Comment:
This `initialize` method is creating/loading train and validation dataset.
Right now this is CIFAR10 so it is fine.
If the dataset size grows larger, our simulator might have problem as we seen in xgboost tree-based with HIGGs dataset @yhwen (in the START_RUN event)

This is just a note.